velox/type/tests/TimestampConversionTest.cpp

```diff
@@ -110,42 +110,43 @@ TEST(DateTimeUtilTest, fromDateStrInvalid) {
 }
 
 TEST(DateTimeUtilTest, castFromDateString) {
-  for (bool nonStandard : {true, false}) {
-    EXPECT_EQ(0, castFromDateString("1970-01-01", nonStandard));
-    EXPECT_EQ(3789742, castFromDateString("12345-12-18", nonStandard));
+  for (bool standard : {true, false}) {
+    EXPECT_EQ(0, castFromDateString("1970-01-01", standard));
+    EXPECT_EQ(3789742, castFromDateString("12345-12-18", standard));
 
-    EXPECT_EQ(1, castFromDateString("+1970-1-2", nonStandard));
-    EXPECT_EQ(1, castFromDateString("+1970-01-2", nonStandard));
-    EXPECT_EQ(1, castFromDateString("+1970-1-02", nonStandard));
+    EXPECT_EQ(1, castFromDateString("+1970-1-2", standard));
```

Comment:
Presto and spark also support this format, can you test this one?
```
presto:default> SELECT cast('1970-01-01 ' as date);
   _col0
------------
 1970-01-01
(1 row)

spark-sql> SELECT cast('1970-01-01 ' as date);
1970-01-01
Time taken: 2.703 seconds, Fetched 1 row(s)
```

Reply:
@jinchengchenghh Thanks. This pattern is not supported in this PR. I'll fix it. Could you help to check whether prefix spaces are valid in presto?

`select cast(' 1970-01-01' as date);`

Reply:
presto:default> select cast(' 1970-01-01' as date);
   _col0
------------
 1970-01-01
(1 row)

Query 20230828_085646_00002_sqyi8, FINISHED, 1 node